src/Cli/dotnet/commands/dotnet-test/Program.cs

```diff
@@ -118,15 +118,21 @@ private static TestCommand FromParseResult(ParseResult result, string[] settings
                 "-nologo"
             };
 
-            msbuildArgs.AddRange(result.OptionValuesToBeForwarded(TestCommandParser.GetCommand()));
+
+            var parsedArgs =
+                    result.OptionValuesToBeForwarded(TestCommandParser.GetCommand()) // all msbuild-recognized tokens
+                    .Concat(result.UnmatchedTokens); // all tokens that the test-parser doesn't explicitly track
```

Comment:
Nit: Indent is too deep

```suggestion
            var parsedArgs =
                result.OptionValuesToBeForwarded(TestCommandParser.GetCommand()) // all msbuild-recognized tokens
                    .Concat(result.UnmatchedTokens); // all tokens that the test-parser doesn't explicitly track
```

Reply:
Or if you prefer to keep `.Concat` on the same indent.

```suggestion
            var parsedArgs =
                result.OptionValuesToBeForwarded(TestCommandParser.GetCommand()) // all msbuild-recognized tokens
                .Concat(result.UnmatchedTokens); // all tokens that the test-parser doesn't explicitly track
```